src/eos-updater-poll-common.c

```diff
@@ -0,0 +1,910 @@
+  status = soup_session_send_message (soup, msg);
+  if (SOUP_STATUS_IS_SUCCESSFUL (status))
+    g_object_get (msg,
+                  SOUP_MESSAGE_RESPONSE_BODY_DATA, &contents,
+                  NULL);
+
+  return g_steal_pointer (&contents);
+}
+
+gboolean
+download_file_and_signature (const gchar *url,
+                             GBytes **contents,
+                             GBytes **signature,
+                             GError **error)
+{
+  g_autoptr(SoupURI) uri = soup_uri_new (url);
+  g_autoptr(SoupURI) sig_uri = NULL;
+
+  if (uri == NULL)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                   "Invalid URL %s", url);
+      return FALSE;
+    }
+
+  sig_uri = get_uri_to_sig (uri);
+  *contents = download_file (uri);
+  *signature = download_file (sig_uri);
+  return TRUE;
+}
+
+gboolean
+get_origin_refspec (OstreeDeployment *booted_deployment,
+                    gchar **out_refspec,
+                    GError **error)
+{
+  GKeyFile *origin;
+  g_autofree gchar *refspec = NULL;
+
+  g_return_val_if_fail (OSTREE_IS_DEPLOYMENT (booted_deployment), FALSE);
+  g_return_val_if_fail (out_refspec != NULL, FALSE);
+
+  origin = ostree_deployment_get_origin (booted_deployment);
+  if (origin == NULL)
+    {
+      const gchar *osname;
+      const gchar *booted;
+
+      osname = ostree_deployment_get_osname (booted_deployment);
+      booted = ostree_deployment_get_csum (booted_deployment);
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND,
+                   "No origin found for %s (%s), cannot upgrade",
+                   osname, booted);
+      return FALSE;
+    }
+
+  refspec = g_key_file_get_string (origin, "origin", "refspec", error);
+  if (refspec == NULL)
+    return FALSE;
+
+  *out_refspec = g_steal_pointer (&refspec);
+  return TRUE;
+}
+
+static void
+get_arm_hw_descriptors (GHashTable *hw_descriptors)
+{
+  g_autoptr(GFile) fp = NULL;
+  g_autofree gchar *fc = NULL;
+
+  fp = g_file_new_for_path (DT_COMPATIBLE);
+  if (g_file_load_contents (fp, NULL, &fc, NULL, NULL, NULL))
+    {
+      g_auto(GStrv) sv = g_strsplit (fc, ",", -1);
+
+      if (sv && sv[0])
+        g_hash_table_insert (hw_descriptors, g_strdup (VENDOR_KEY),
+                             g_strdup (g_strstrip (sv[0])));
+      if (sv && sv[1])
+        g_hash_table_insert (hw_descriptors, g_strdup (PRODUCT_KEY),
+                             g_strdup (g_strstrip (sv[1])));
+    }
+}
+
+static void
+get_x86_hw_descriptors (GHashTable *hw_descriptors)
+{
+  guint i;
+
+  for (i = 0; dmi_attributes[i]; i++)
+    {
+      g_autofree gchar *path = NULL;
+      g_autoptr(GFile) fp = NULL;
+      g_autofree gchar *fc = NULL;
+      gsize len;
+
+      path = g_strconcat (DMI_PATH, dmi_attributes[i], NULL);
+      fp = g_file_new_for_path (path);
+      if (g_file_load_contents (fp, NULL, &fc, &len, NULL, NULL))
+        {
+          if (len > 128)
+            fc[128] = '\0';
```

Comment:
No idea, it was like that before, see https://github.com/endlessm/eos-updater/blob/master/src/eos-updater-util.c#L269